docs/rest-server/API.md

```diff
@@ -62,914 +62,954 @@ Configure the rest server port in [services-configuration.yaml](../../examples/c
+
+*Response if succeeded*
+```
+Status: 200
+
+{
+  "containers": [
+    {
+      "id": "<container id>",
+      "sshIp": "<ip to access the container's ssh service>",
+      "sshPort": "<port to access the container's ssh service>"
+    },
+    ...
+  ],
+  "keyPair": {
+    "folderPath": "HDFS path to the job's ssh folder",
+    "publicKeyFileName": "file name of the public key file",
+    "privateKeyFileName": "file name of the private key file",
+    "privateKeyDirectDownloadLink": "HTTP URL to download the private key file"
+  }
+}
+```
+
+*Response if the job does not exist*
+```
+Status: 404
+
+{
+  "code": "NoJobError",
+  "message": "Job $jobname is not found."
+}
+```
+
+*Response if the job SSH info does not exist*
+```
+Status: 404
+
+{
+  "code": "NoJobSshInfoError",
+  "message": "SSH info of job $jobname is not found."
+}
+```
+
+*Response if a server error occured*
+```
+Status: 500
+
+{
+  "code": "UnknownError",
+  "message": "*Upstream error messages*"
+}
+```
+
+### `PUT user/:username/jobs/:jobName/executionType`
+
+Start or stop a job.
+
+*Request*
+```
+PUT /api/v1/user/:username/jobs/:jobName/executionType
+Authorization: Bearer <ACCESS_TOKEN>
+```
+
+*Parameters*
+```
+{
+  "value": "START" | "STOP"
+}
+```
+
+*Response if succeeded*
+```
+Status: 200
+
+{
+  "message": "execute job $jobName successfully"
+}
+```
+
+*Response if the job does not exist*
+```
+Status: 404
+
+{
+  "code": "NoJobError",
+  "message": "Job $jobname is not found."
+}
+```
+
+*Response if a server error occured*
+```
+Status: 500
+
+{
+  "code": "UnknownError",
+  "message": "*Upstream error messages*"
+}
+```
+
+### `GET virtual-clusters`
+
+Get the list of virtual clusters.
+
+*Request*
+```
+GET /api/v1/virtual-clusters
+```
+
+*Response if succeeded*
+```
+Status: 200
+
+{
+  "vc1":
+  {
+  }
+  ...
+}
+```
+
+*Response if a server error occured*
+```
+Status: 500
+
+{
+  "code": "UnknownError",
+  "message": "*Upstream error messages*"
+}
+```
+
+### `GET virtual-clusters/:vcName`
+
+Get virtual cluster status in the system.
+
+*Request*
+```
+GET /api/v1/virtual-clusters/:vcName
+```
+
+*Response if succeeded*
+```
+Status: 200
+
+{
+  //capacity percentage this virtual cluster can use of entire cluster
+  "capacity":50,
+  //max capacity percentage this virtual cluster can use of entire cluster
+  "maxCapacity":100,
+  // used capacity percentage this virtual cluster can use of entire cluster
+  "usedCapacity":0,
+  "numActiveJobs":0,
+  "numJobs":0,
+  "numPendingJobs":0,
+  "resourcesUsed":{
+   "memory":0,
+   "vCores":0,
+   "GPUs":0
+  },
+  "state":"running"
+}
+```
+
+*Response if the virtual cluster does not exist*
+```
+Status: 404
+
+{
+  "code": "NoVirtualClusterError",
+  "message": "Virtual cluster $vcname is not found."
+}
+```
+
+*Response if a server error occured*
+```
+Status: 500
+
+{
+  "code": "UnknownError",
+  "message": "*Upstream error messages*"
+}
+```
+
+### `PUT virtual-clusters/:vcName` (administrator only)
```

Comment:
Should be removed after this line, they are new API in v0.9

Reply:
resolved